jbpm-event-emitters/jbpm-event-emitters-kafka/src/main/java/org/jbpm/event/emitters/kafka/KafkaEventEmitter.java

```diff
@@ -0,0 +1,142 @@
+                                "yyyy-MM-dd'T'HH:mm:ss.SSSZ"))))
+                .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+                .configure(MapperFeature.PROPAGATE_TRANSIENT_MARKER, true);
+    }
+
+    public void deliver(Collection<InstanceView<?>> data) {
+        // no-op
+    }
+
+    public void apply(Collection<InstanceView<?>> data) {
+        if (data == null || data.isEmpty()) {
+            return;
+        }
+
+        for (InstanceView<?> view : data) {
+            String processId;
+            long processInstanceId;
+            String type;
+            String topic;
+            if (view instanceof ProcessInstanceView) {
+                ProcessInstanceView processInstanceView = (ProcessInstanceView) view;
+                topic = "processes";
+                type = "process";
+                processInstanceId = processInstanceView.getId();
+                processId = processInstanceView.getProcessId();
+            } else if (view instanceof TaskInstanceView) {
+                TaskInstanceView taskInstanceView = (TaskInstanceView) view;
+                topic = "tasks";
+                type = "task";
+                processInstanceId = taskInstanceView.getProcessInstanceId();
+                processId = taskInstanceView.getProcessId();
+            } else if (view instanceof CaseInstanceView) {
+                CaseInstanceView caseInstanceView = (CaseInstanceView) view;
+                topic = "cases";
+                type = "case";
+                processInstanceId = caseInstanceView.getId();
+                processId = caseInstanceView.getCaseDefinitionId();
+            } else {
+                throw new UnsupportedOperationException("view " + view.getClass());
+            }
+            try {
+                producer.send(new ProducerRecord<>(getTopic(topic), mapper.writeValueAsBytes(
+                        new CloudEventSpec1(type, String.format(SOURCE_FORMATTER, processId, processInstanceId),
+                                view))));
+            } catch (JsonProcessingException e) {
+                throw new IllegalArgumentException("cannot convert " + view + " to byte[]", e);
+            }
+        }
+    }
+
+    private static String getTopic(String eventType) {
+        return System.getProperty("org.jbpm.event.emitters.kafka.topic." + eventType, "jbpm-" + eventType + "-events");
+    }
+
+    public void drop(Collection<InstanceView<?>> data) {
+        // no-op
+    }
+
+    @Override
+    public void close() {
+        producer.close();
+    }
+
+    @Override
+    public EventCollection newCollection() {
+        return new BaseEventCollection();
+    }
+
+    private static Map<String, Object> getConfigs() {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, System.getProperty(
+                "org.jbpm.event.emitters.kafka.boopstrap.servers", "localhost:9092"));
```

Comment:
org.kie.jbpm.emitter.kafka.bootstrap.servers. 
That is the standard.

Reply:
Changed